bundles/org.openhab.binding.hapero/src/main/java/org/openhab/binding/hapero/internal/device/Device.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hapero.internal.device;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Device} class contains the raw
+ * data received from the heating control system and
+ * conversion functions.
+ *
+ * @author Daniel Walter - Initial contribution
+ */
+@NonNullByDefault
+public class Device {
+    private final Logger logger = LoggerFactory.getLogger(Device.class);
+    /** Raw data stream from the device */
+    protected String[] dataItems = {};
+
+    /**
+     * Constructor
+     *
+     * @param items List of data items as Strings
+     */
+    public Device(String[] items) {
+        dataItems = items;
+    }
+
+    /**
+     * Return the data item at the given index
+     *
+     * @return A String containing the data item at the Index or "" if the Index is invalid
+     */
+    public String getDataItem(int index) {
+        if (index >= 0 && index < dataItems.length) {
+            return dataItems[index];
+        } else {
+            logger.warn("getDataItem: Invalid Index {}.", index);
+            return "";
+        }
+    }
+
+    /**
+     * @param dataItems
+     */
+    public void setDataItems(String[] dataItems) {
+        this.dataItems = dataItems;
+    }
+
+    /**
+     * Returns the value of the dataItem at Index as a Float
+     *
+     * @param index Index of the dataItem to return
+     * @return Float or null if the index is invalid or the dataItem can not be converted to Float
+     */
+    public @Nullable Float getFloat(int index) {
+        Float result = null;
+
+        if (index >= 0 && index < dataItems.length) {
```

Comment:
This control is already present in `getDataItem`, maybe you could directly call `Float.valueOf(getDataItem[index])`

Reply:
Yes, but if a invalid index is requested, getDataItem would return "" which then would throw a NumberFormatException, so you would always get 2 error messages (invalid index + the exception). Not a big deal but also not helpful. I could of course test the result for "" before trying to parse it but that additional code would introduce another "if" instead of the previous one. Or is there a nice Java trick to do it ?